File: SMP/SMP.Web/Modules/ServerTypes/Productos/TblproductosDidiImportExcelForm.ts
import { ImageUploadEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface TblproductosDidiImportExcelForm {
    FileName: ImageUploadEditor;
}

export class TblproductosDidiImportExcelForm extends PrefixedContext {
    static formKey = 'Productos.ImportExcel.TblproductosDidi';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TblproductosDidiImportExcelForm.init)  {
            TblproductosDidiImportExcelForm.init = true;

            var w0 = ImageUploadEditor;

            initFormType(TblproductosDidiImportExcelForm, [
                'FileName', w0
            ]);
        }
    }
}

File: SMP/SMP.Web/Modules/Productos/ImportExcel/TblproductosDidiImportExcelForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace SMP.Productos;

[FormScript("Productos.ImportExcel.TblproductosDidi")]
//[BasedOnRow(typeof(TblproductosDidiRow), CheckNames = false)]
[BasedOnRow(typeof(TblproductosDidiRow), CheckNames = false)]
public class TblproductosDidiImportExcelForm
{
    //[ReadOnly(true)]
    //public Int32 IdCatCategoriaPlanograma { get; set; }
    //[Hidden]
    //public Int32 IdOrigen { get; set; }
    //[Hidden]
    //public Int32 idArticulo { get; set; }
    //public String VchCodCategoriaPlanograma { get; set; }
    //public String VchDesCategoriaPlanograma { get; set; }

    //public String VchComentarios { get; set; }
    //[BooleanEditor]
    //public Int32 IEliminado { get; set; }
    [FileUploadEditor, Required]
    public string FileName { get; set; }
}